vowpalwabbit/parser.cc

```diff
@@ -484,6 +484,7 @@ void enable_sources(vw& all, bool quiet, size_t passes)
     // background process (if foreground is not set)
     if (!all.opts_n_args.vm.count("foreground"))
     {
+      //FIXME switch to posix_spawn
```

Comment:
I'm unfamiliar with posix_spawn.  What's needed?

Reply:
Could we use a std::thread instead so that the code is portable?

Reply:
@JohnLangford daemon is deprecated on OSX. 
Additionally posix_spawn gives fine grained control over the signal mask and all the process IDs.

It's not clear to me if those are features needed by VW particular.

@jackgerrits both daemon and posix_spawn create a new process and the former exits the parent. Threads can't provide that.